### PR TITLE
ROS-40 Add OpenAPI specs

### PR DESCRIPTION
**Pull request by apuntamb**
Adding the OpenAPI specifications for the ROS v0 api. The following screenshot depicts Round 1 of the openapi UI as this is currently in WIP phase. Please feel free to add review comments and suggestions.
Also, adding @beav in the loop for review.

openapi_spec_local

### CODE REVIEW

**Comment by beav** [comment]
Overall this looks good from my perspective. Thanks!

**Comment by apuntamb** [comment]
> Overall this looks good from my perspective. Thanks!

Could you suggest a better way of integrating this spec file with ros `flask_restful` app?

File: ros/openapi/openapi.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        "/systems": {
+            "get": {
+                "summary": "Show all systems from host inventory having performance profile",
+                "description": "Show all systems from host inventory having performance profile",
+                "operationId": "listSystems",
+                "parameters": [
+                    {
+                        "name": "limit",
+                        "in": "query",
+                        "description": "Limit for paging, set -1 to return all",
```
Comment (kgaikwad) [inline ros/openapi/openapi.json:46]:
```suggestion
                        "description": "The numbers of items to return, set -1 to return all",
```

File: ros/openapi/openapi.json [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                        "schema": {
+                            "type": "integer"
+                        }
+                    },
+                    {
+                        "name": "offset",
+                        "in": "query",
+                        "description": "Offset for paging",
```
Comment (kgaikwad) [inline ros/openapi/openapi.json:54]:
```suggestion
                        "description": "The number of items to skip before starting to collect the result set",
```

File: ros/openapi/openapi.json [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/systems": {
+            "get": {
+                "summary": "Show all systems from host inventory having performance profile",
```
Comment (kgaikwad) [inline ros/openapi/openapi.json:39]:
```suggestion
                "summary": "Lists all systems from cloud provider having performance profile",
```

File: ros/openapi/openapi.json [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                "description": "Show all systems from host inventory having performance profile",
```
Comment (kgaikwad) [inline ros/openapi/openapi.json:40]: ditto

File: ros/openapi/openapi.json [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                    {
+                        "name": "offset",
+                        "in": "query",
+                        "description": "Offset for paging",
+                        "schema": {
+                            "type": "integer"
+                        }
+                    },
+                    {
+                        "name": "sort",
```
Comment (kgaikwad) [inline ros/openapi/openapi.json:60]: As per API design guidelines - https://platform-docs.cloud.paas.psi.redhat.com/rest/sorting.html, for sorting it is recommended to set `sort_by` query parameter. 
I don't want to block the PR as it's in initial state but thought to share for reference.

**Review by kgaikwad — CHANGES_REQUESTED** [review]
@apuntamb, requested few inline changes. 
Otherwise, as an initial version it looks good to me.

File: ros/openapi/openapi.json (hunk 5 above)
Comment (apuntamb) [reply ros/openapi/openapi.json:60]: Yeah sure, let's consider this for refactoring when openapi specs would change after #18, makes sense?

File: ros/openapi/openapi.json (hunk 2 above)
Comment (apuntamb) [reply ros/openapi/openapi.json]: done

File: ros/openapi/openapi.json (hunk 3 above)
Comment (apuntamb) [reply ros/openapi/openapi.json]: done

File: ros/openapi/openapi.json (hunk 4 above)
Comment (apuntamb) [reply ros/openapi/openapi.json]: done

**Comment by apuntamb** [comment]
Latest output after applying pagination related changes:
```json

    {
      "data": [
        {
          "fqdn": "string",
          "display_name": "string",
          "id": "string",
          "recommendation_count": 0,
          "state": "string",
          "performance_score": {
            "memory_score": 0,
            "cpu_score": 0,
            "io_score": 0
          },
          "facts": [
            {}
          ]
        }
      ],
      "links": {
        "first": "string",
        "last": "string",
        "next": "string",
        "previous": "string"
      },
      "meta": {
        "limit": 0,
        "offset": 0,
        "count": 0
      }
    }

```

File: ros/openapi/openapi.json (hunk 5 above)
Comment (apuntamb) [reply ros/openapi/openapi.json]: since we do not have sorting/searching/filtering options as of now, eliminating this part.

**Comment by apuntamb** [comment]
Merged, thanks @kgaikwad @beav for quick review!

**Pull request merged**